app/models/fe/question_set.rb

```diff
@@ -106,7 +106,9 @@ def posted_values(param)
         end
       elsif param.kind_of?(Hash)
         # from Hash with multiple answers per question
-        values = param.values.map {|v| CGI.unescape(v)}
+        # If value is also a hash, use the value hash without escaping or anything,
+        # so that custom elements can be implemented by handling set_response.
+        values = param.values.map {|v| v.is_a?(Hash) ? v : CGI.unescape(v)}
```

Comment:
No, it's not a security concern as far as I can figure.  Think of this more as just passing on raw data to the enclosing app, and it's up to the enclosing app to do what they want with it.  ie. in save_response like https://github.com/CruGlobal/pr/blob/e4101c9b4ee7089d9c040c0f3c46115596b9fbd8/app/models/fe/cmo.rb#L6

Also, back when this was built, rails didn't have the "escape by default" policy when outputting strings, so we were more susceptible to cross-site scripting attacks.  As it is, I'm not even sure we need GCI.unescape(v), but I left it in in case some old apps depend on it.  Just to be sure, I tested saving a CMO answer as "<script>alert('hi');</script>" and it escaped everything properly.